DSC Resource/JustEnoughAdministration/JustEnoughAdministration.psm1

```diff
@@ -490,17 +490,68 @@ class JeaEndpoint
         return $true
     }
 
-    ## A simple comparison for complex objects used in JEA configurations.
-    ## We don't need anything extensive, as we should be the only ones changing them.
-    hidden [bool] ComplexObjectsEqual($object1, $object2)
+    ## A comparison for complex objects used in JEA configurations.
```

Comment:
I am not very familiar with this code.  Please add comments that provide examples of the types of complex objects being compared.  Also please add tests to validate the compare function.

My understanding is that we only compare hash tables, and hash tables within hash tables.  If so I wonder if we need to convert to json at all?  Instead just compare hash table key/values.  If the value is another hash table, then recurse.  
Something like:
```powershell
foreach ($key in $ht1.Keys) { CompareHashValues($ht1[$key], $ht2[$key]) }
```
Will that work?